core/src/main/java/org/elasticsearch/index/mapper/DateFieldMapper.java

```diff
@@ -490,8 +490,10 @@ protected void parseCreateField(ParseContext context, List<IndexableField> field
 
     @Override
     protected void doMerge(Mapper mergeWith, boolean updateAllTypes) {
+        final DateFieldMapper other = (DateFieldMapper) mergeWith;
+        if (!this.fieldType().equals(other.fieldType()))
+            throw new IllegalArgumentException("date field cannot be updated");
```

Comment:
Could you only check the `format` property and make the error message more specific?